### PR TITLE
Micronaut 3

### PR DESCRIPTION
**Pull request by ryandens**
## Description
Note, this change bumps the version of Hypertrace from `1.1.X` to `1.2.X` (felt appropriate to me)

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: instrumentation/netty/netty-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/netty/v4_0/NettyChannelPipelineInstrumentation.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ public static void addHandler(
         @Advice.Enter int callDepth,
         @Advice.This ChannelPipeline pipeline,
         @Advice.Argument(2) ChannelHandler handler) {
-      if (callDepth > 0) {
```
Comment (shashank11p) [inline instrumentation/netty/netty-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/netty/v4_0/NettyChannelPipelineInstrumentation.java:85]: Why do we want to increase the call depth here?

File: instrumentation/netty/netty-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/netty/v4_0/client/HttpClientResponseTracingHandler.java [hunk 2]
```diff
@@ -104,7 +104,7 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
       span.setAttribute(SemanticAttributes.HTTP_STATUS_CODE, httpResponse.getStatus().code());
       span.setStatus(HttpStatusConverter.statusFromHttpStatus(httpResponse.getStatus().code()));
     }
-    if (msg instanceof FullHttpMessage || msg instanceof LastHttpContent) {
```
Comment (shashank11p) [inline instrumentation/netty/netty-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/netty/v4_0/client/HttpClientResponseTracingHandler.java:107]: Do we not want to check the FullHttpMessage ever now?

Comment (ryandens) [reply instrumentation/netty/netty-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/netty/v4_0/client/HttpClientResponseTracingHandler.java:107]: In this case, `FullHttpMessage` extends `LastHttpContent` so this check was redundant

File: instrumentation/netty/netty-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/netty/v4_0/NettyChannelPipelineInstrumentation.java (hunk 1 above)
Comment (ryandens) [reply instrumentation/netty/netty-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/netty/v4_0/NettyChannelPipelineInstrumentation.java:85]: This was an odd one, but I found this was  a niche bug I found where sometimes the handler's call depth was one but the channel pipeline had not yet had its tracing handlers replaced

**Review by shashank11p — APPROVED** [review]

File: instrumentation/netty/netty-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/netty/v4_0/NettyChannelPipelineInstrumentation.java (hunk 1 above)
Comment (ryandens) [reply instrumentation/netty/netty-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/netty/v4_0/NettyChannelPipelineInstrumentation.java]: Shashank and I talked offline about this one - we only want to change this number for netty 4.1 due to the different way the call depth map is constructed

**Pull request merged**